emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/preader/ScanRecordReadRequest.java

```diff
@@ -54,4 +63,44 @@ protected PageResults<Map<String, AttributeValue>> fetchPage(RequestLimit lim) {
         consumedCapacityUnits,
         retries);
   }
+
+  /**
+   * Set a Scan query filter to skip expired records if the configuration
+   * provides the TTL attribute name
+   */
+  Optional<DynamoDBQueryFilter> queryFilter() {
+    Optional<String> maybeTtlAttribute =
+        Optional.ofNullable(context.getConf().get(DynamoDBConstants.TTL_ATTRIBUTE_NAME));
+    return maybeTtlAttribute.map(attributeName -> {
+      long now = Instant.now().getEpochSecond();
+      DynamoDBQueryFilter filter = new DynamoDBQueryFilter();
+      filter.addScanFilter(new DynamoDBFilter() {
+        @Override
+        public String getColumnName() {
+          return attributeName;
+        }
+
+        @Override
+        public String getColumnType() {
+          throw new Error();
+        }
+
+        @Override
+        public DynamoDBFilterOperator getOperator() {
+          throw new Error();
+        }
+
+        @Override
+        public Condition getDynamoDBCondition() {
+          return Condition
+              .builder()
+              .comparisonOperator(ComparisonOperator.GT)
+              .attributeValueList(AttributeValue.fromN(String.valueOf(now)))
+              .build();
```

Comment:
Note that we are using an old API of the SDK here. Eventually, we could replace it with the more recent “filter expressions”:

~~~ java
Map<String, AttributeValue> expressionAttributeValues = new HashMap<>();
        expressionAttributeValues.put(":currentTime", AttributeValue.builder().n(String.valueOf(Instant.now().getEpochSecond())).build());

ScanRequest scanRequest = ScanRequest.builder()
        .tableName(tableName)
        .filterExpression("attribute_not_exists(" + ttlAttributeName + ") OR " + ttlAttributeName + " > :currentTime")
        .expressionAttributeValues(expressionAttributeValues)
        .build();
~~~

But this would require more changes in emr-dynamodb-connector, since it currently only support `scanFilter`, not `filterExpression` and `expressionAttributeValues`.